covalent_ec2_plugin/ec2.py

```diff
@@ -175,10 +201,8 @@ async def _run_async_subprocess(self, cmd: List[str], cwd=None, log_output: bool
 
         return proc, stdout, stderr
 
-    def _get_tf_statefile_path(self, task_metadata: Dict) -> str:
-        state_file = (
-            f"{self._TF_DIR}/ec2-{task_metadata['dispatch_id']}-{task_metadata['node_id']}.tfstate"
-        )
+    def _get_tf_statefile_path(self) -> str:
+        state_file = f"{self._TF_DIR}/terraform.tfstate"
```

Comment:
The task metadata should be used here because not all the tasks which will use this executor will have the same statefile. The statefile is (or should be) created particularly for each dispatch-id/node-id combination.